code/modules/projectiles/guns/specialist.dm

```diff
@@ -600,39 +574,65 @@
-	playsound(user.loc, cocked_sound, 25, 1)
-	last_fired = world.time
-	visible_message(span_danger("[user] fired a grenade!"))
-	to_chat(user, span_warning("You fire [src]!"))
-	var/obj/item/explosive/grenade/F = grenades[1]
-	grenades -= F
-	F.loc = user.loc
-	F.throw_range = 20
-	if(F?.loc) //Apparently it can get deleted before the next thing takes place, so it runtimes.
-		log_explosion("[key_name(user)] fired a grenade [F] from [src] at [AREACOORD(user.loc)].")
-		log_combat(user, src, "fired a grenade [F] from [src]")
-		F.det_time = min(10, F.det_time)
-		F.launched = TRUE
-		F.activate(user)
-		F.throwforce += F.launchforce //Throws with signifcantly more force than a standard marine can.
-		F.throw_at(target, 20, 3, user)
-		playsound(F.loc, fire_sound, 50, 1)
-		if(fire_animation)
-			flick("[fire_animation]", src)
-
-/obj/item/weapon/gun/launcher/m92/get_ammo_type()
-	if(length(grenades) == 0)
+/obj/item/weapon/gun/launcher/get_ammo_type()
+	if(!length(grenades))
 		return list("empty", "empty")
 	else
 		var/obj/item/explosive/grenade/F = grenades[1]
 		return list(F.hud_state, F.hud_state_empty)
 
-/obj/item/weapon/gun/launcher/m92/get_ammo_count()
-	return length(grenades)
+/obj/item/weapon/gun/launcher/get_ammo_count()
+	if(!length(grenades))
+		return 0
+	else
+		return length(grenades)
+
+/obj/item/weapon/gun/launcher/m92
+	name = "\improper T-26 grenade launcher"
+	desc = "A heavy, 6-shot grenade launcher used by the TerraGov Marine Corps for area denial and big explosions."
+	icon_state = "m92"
+	item_state = "m92"
+	max_shells = 6 //codex
+	wield_delay = 0.6 SECONDS
+	aim_slowdown = 1
+	attachable_allowed = list(/obj/item/attachable/magnetic_harness, /obj/item/attachable/scope/mini)
+	attachable_offset = list("muzzle_x" = 33, "muzzle_y" = 18,"rail_x" = 14, "rail_y" = 22, "under_x" = 19, "under_y" = 14, "stock_x" = 19, "stock_y" = 14)
+	fire_delay = 1.8 SECONDS
+	max_grenades = 6
+	var/datum/effect_system/smoke_spread/smoke
+
+/obj/item/weapon/gun/launcher/m92/Initialize()
+	. = ..()
+	for(var/i in 1 to max_grenades)
+		grenades += new /obj/item/explosive/grenade(src)
+
+/obj/item/weapon/gun/launcher/m92/update_icon(mob/user)
+	update_item_state(user)
+	update_mag_overlay(user)
+
+/obj/item/weapon/gun/launcher/m92/examine_ammo_count(mob/user)
+	if(!length(grenades) || (get_dist(user, src) > 2 && user != loc))
+		return
+	to_chat(user, span_notice("It is loaded with <b>[length(grenades)] / [max_grenades]</b> grenades."))
+
+/obj/item/weapon/gun/launcher/m92/Fire()
+	if(CHECK_BITFIELD(flags_gun_features, GUN_DEPLOYED_FIRE_ONLY) && !CHECK_BITFIELD(flags_item, IS_DEPLOYED))
+		to_chat(gun_user, span_notice("You cannot fire [src] while it is not deployed."))
+		return
+	if(CHECK_BITFIELD(flags_gun_features, GUN_IS_ATTACHMENT) && !master_gun && CHECK_BITFIELD(flags_gun_features, GUN_ATTACHMENT_FIRE_ONLY))
+		to_chat(gun_user, span_notice("You cannot fire [src] without it attached to a gun!"))
+		return
+	. = ..()
```

Comment:
`return ..()`
But like i said before you should just put these checks back at the top